### PR TITLE
block access list

### PR DESCRIPTION
**Pull request by g11in**
a proposal to build a block access list and include its fingerprint in block header
https://ethresear.ch/t/block-access-list-v0-1/9505

### CODE REVIEW

**Comment by alita-moore** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):


	 - File with name EIPS/eip-block_access_list.md is new and new files must be reviewed
	 - Filename EIPS/eip-block_access_list.md is not in EIP format 'EIPS/eip-####.md'

File: EIPS/eip-block_access_list.md
Comment (MicahZoltu) [inline EIPS/eip-block_access_list.md:2]:
```suggestion
eip: 3584
```

**Comment by MicahZoltu** [comment]
@g11in Please see https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for details on the format, sections, and what each section of an EIP should contain.

**Comment by g11in** [comment]
> 
> 
> @g11in Please see https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for details on the format, sections, and what each section of an EIP should contain.

Hi Micah, thanks for the response. I did see the template but i thought the sections are more of a guidance. let me see how I can rearrange. :)

File: EIPS/eip-block_access_list.md
Comment (g11in) [reply EIPS/eip-block_access_list.md:2]: may be just a very naive question, should I also change the file name too eip-3584.md format?

Comment (MicahZoltu) [reply EIPS/eip-block_access_list.md:2]: Yes, sorry.  I forgot to mention that.  Filename should be `eip-3584.md`.

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:104]: This should not be included.  Core EIPs focus on consensus changes, while changes to JSON-RPC happen in another repository and usually *after* the core EIP is finalized.
```suggestion
```

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:115]: This should be removed or moved to the rationale.  The specification section should limit itself to only the things that must be implemented by someone wanting to adopt this specification.  It may make sense to mention this in the Rationale if certain design choices were made specifically to enable future expansion.  Alternatively, consider writing another EIP that depends on this one which defines next step, or keep such information in another location.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Blockers for merging are:
* use relative links when referencing other EIPs.
* Not skipping nesting levels (`###` vs `####`)
* `Security Considerations` section is required.

I recommend reviewing all of the other feedback as well, but it won't block merging as a draft.

File: EIPS/eip-3584.md
Comment (g11in) [reply EIPS/eip-3584.md]: moved to rationale.

**Comment by g11in** [comment]
thanks @MicahZoltu  for the feedback, committed the modifications.

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:37]: Mixed tabs and spaces (looks like spaces everywhere else, but there is a tab leading this line).

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:106]: Tabs/spaces again.

**Review by MicahZoltu — APPROVED** [review]

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:78]:
```suggestion
All sorting is in increasing order.
```
Spell checker doesn't like `sortings`.

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:91]:
```suggestion
It is specified to be sorted in lexicographic ordering or integer sorting wherever applicable and specified. Sorting with respect to access time was considered but didn't seem to provide any additional benefit at the cost of adding implementation complexity and bookkeeping.
```

File: EIPS/eip-3584.md
Comment (MicahZoltu) [inline EIPS/eip-3584.md:112]:
```suggestion
and then get to define the a canonical specification for building the fingerprint.
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Spelling errors need to be fixed before we can merge.

**Review by MicahZoltu — APPROVED** [review]
I went ahead and applied the spelling and indentation fixes since they seemed pretty non-contentious.  Once CI passes this can be merged.

**Comment by g11in** [comment]
> 
> 
> I went ahead and applied the spelling and indentation fixes since they seemed pretty non-contentious. Once CI passes this can be merged.

thanks you!

**Pull request merged**
